src/broyden.jl

```diff
@@ -0,0 +1,170 @@
+# based on 
+# A QUASI-NEWTON METHOD FOR SOLVING SMALL NONLINEAR SYSTEMS OF ALGEBRAIC EQUATIONS
+# JEFFREY MARTIN
+# http://web.mit.edu/jmartin3/Public/Project.pdf
+macro broydentrace(stepnorm)
+    quote
+        if tracing
+            dt = Dict()
+            if extended_trace
+                dt["x"] = copy(x)
+                dt["f(x)"] = copy(fvec)
+                dt["g(x)"] = copy(fjac)
+            end
+            update!(tr,
+                    it,
+                    maximum(abs(fvec)),
+                    $stepnorm,
+                    dt,
+                    store_trace,
+                    show_trace)
+        end
+    end
+end
+
+
+const τ = 0.5
+const σ1 = 1e-8
+const σ2 = 1e-8
+const ρ = 1.0 - 1e-8
+const η = 1e-8
+const restol = 1e-6
+const imax = 10
+function broyden_{T}(df::AbstractDifferentiableMultivariateFunction,
+                   initial_x::Vector{T},
+                   xtol::T,
+                   ftol::T,
+                   iterations::Integer,
+                   store_trace::Bool,
+                   show_trace::Bool,
+                   extended_trace::Bool)
+
+    x = copy(initial_x)
+    nn = length(x)
+    xold = fill(convert(T, NaN), nn)
+    fvec = Array(T, nn)
+    fvecold = similar(fvec)
+    fjac = alloc_jacobian(df, T, nn)
+    scaleinvfjac = alloc_jacobian(df, T, nn)
+    oldinvfjac = alloc_jacobian(df, T, nn)
+    minvjΔf = similar(xold)
+    p = Array(T, nn)
+    g = Array(T, nn)
+    gr = Array(T, nn)
+
+    # Count function calls
+    f_calls::Int, g_calls::Int = 0, 0
+
+    df.fg!(x, fvec, fjac)
+    f_calls += 1
+    g_calls += 1
+
+    check_isfinite(fvec)
+    normfvec = norm(fvec)
+    normfvecold = normfvec
+    normfvecoldold = normfvec
+
+    it = 0
+    x_converged, f_converged, converged = assess_convergence(x, xold, fvec, xtol, ftol)
+
+    computejac = true
+
+    tr = SolverTrace()
+    tracing = store_trace || show_trace || extended_trace
+    @broydentrace convert(T, NaN)
+
+
+    while !converged && it < iterations
+        it += 1
+        if computejac
+            if it > 1
+                df.fg!(x, fvec, fjac)
+                g_calls += 1
+            end
+            try
+                p = fjac \ fvec
+                scale!(p, -1)
+            catch e
+                if isa(e, Base.LinAlg.LAPACKException)
+                    fjac2 = Ac_mul_B(fjac, fjac)
+                    lambda = convert(T, 1e6) * sqrt(nn * eps()) * norm(fjac2, 1)
+                    g = Ac_mul_B(fjac, fvec)
+                    p = (fjac2 + lambda * eye(nn)) \ g
+                    scale!(p, -1)
```

Comment:
What's going on here?  You are squaring the condition number by squaring the matrix.